components/cdc/src/endpoint.rs

```diff
@@ -393,30 +407,52 @@ impl<T: RaftStoreRouter> Endpoint<T> {
         }
     }
 
-    fn on_min_ts(&mut self, min_ts: TimeStamp) {
-        for delegate in self.capture_regions.values_mut() {
+    fn on_min_ts(&mut self, region_id: u64, min_ts: TimeStamp) {
+        if let Some(delegate) = self.capture_regions.get_mut(&region_id) {
             delegate.on_min_ts(min_ts);
         }
-        self.register_min_ts_event();
     }
 
     fn register_min_ts_event(&self) {
         let timeout = self.timer.delay(self.min_ts_interval);
         let tso = self.pd_client.get_tso();
         let scheduler = self.scheduler.clone();
+        let raft_router = self.raft_router.clone();
+        let regions: Vec<u64> = self.capture_regions.keys().copied().collect();
         let fut = tso.join(timeout.map_err(|_| unreachable!())).then(
             move |tso: pd_client::Result<(TimeStamp, ())>| {
                 // Ignore get tso errors since we will retry every `min_ts_interval`.
                 let (min_ts, _) = tso.unwrap_or((TimeStamp::default(), ()));
-                match scheduler.schedule(Task::MinTS { min_ts }) {
-                    Ok(_) | Err(ScheduleError::Stopped(_)) => Ok(()),
-                    // Must schedule `MinTS` event otherwise resolved ts can not
+                for region_id in regions {
+                    let scheduler_clone = scheduler.clone();
+                    if let Err(e) = raft_router.significant_send(
+                        region_id,
+                        SignificantMsg::LeaderCallback(Callback::Read(Box::new(move |resp| {
```

Comment:
It may consume a lot of CPU, consider a TiKV has 10,000 leaders, it will be 10k op/s.

Is it possible to hold a remote lease instead?

Reply:
We need to request a read index for checking leadership after the lease expired, so I think the most simple way is reusing the code of read.